pkg/endpoint/policy.go

```diff
@@ -590,12 +592,19 @@ func (e *Endpoint) regeneratePolicy(owner Owner, opts models.ConfigurationMap) (
 
 	needToRegenerateBPF := optsChanged || policyChanged || e.nextPolicyRevision > e.policyRevision
 
+	regenerateEnd := time.Since(regenerateStart) / time.Second
```

Comment:
You should put this code into a `defer` statement, in order to not ignore regenerations that fail.

Reply:
sure, as described before in the endpoint case, there seems to be locking involved while accessing the `e.getLogger()`, but it can be done here for sure.

Reply:
done